packages/forms/resources/views/components/checkbox-list.blade.php

```diff
@@ -13,32 +13,70 @@
     :state-path="$getStatePath()"
 >
     <div x-data="{
+
         areAllCheckboxesChecked: false,
 
+        checkboxListOptions: Array.from($root.querySelectorAll('.filament-forms-checkbox-list-component-option-label')),
+
+        search: '',
+
+        visibleCheckboxListOptions: [],
+
         init: function () {
             this.checkIfAllCheckboxesAreChecked()
 
             Livewire.hook('message.processed', () => {
                 this.checkIfAllCheckboxesAreChecked()
```

Comment:
Any particular reason why `this.updateVisibleCheckboxListOptions()` wasn't called here? Where having issue when we load the options dynamically. 

```php
Forms\Components\CheckboxList::make('states')
    ->options(
        fn (Closure $get) => ($country = $get('country'))
            ? State::whereCountryId($country)->pluck('name', 'id')
            : []
    )
```

It has been observed that the DOM is getting updated but the option are hidden.

Reply:
Feel free to PR a fix.

Reply:
@danharrin, might have to check if this still works with the Livewire hooks in v3.